Initial installation guide setup:

solipet opened the pull request:
- adds mac os x installation guide
- adds assets/images dir
- adds include for figures

solipet committed:
Initial installation guide setup:
- adds mac os x installation guide
- adds assets/images dir
- adds include for figures

alimi commented on _installation/mac_os.md:
```diff
@@ -0,0 +1,118 @@
+
+If you see the output above, rails is running and you can point your browser at http://localhost:3000 and you should see this:
+
+{% include figure.html url="/assets/images/youre_on_rails.png" description="Yay! You're on Rails!" %}
```
This is a good idea for dealing with images, but could we use the native Markdown syntax instead?

```
![Alt text](/path/to/img.jpg "Optional title")
```

https://daringfireball.net/projects/markdown/syntax#img

alimi commented on installation.md:
```diff
@@ -0,0 +1,35 @@
+---
+# You don't need to edit this file, it's empty on purpose.
```
This comment seems out of place. Especially since the file isn't empty 😉

solipet replied:
hehe, I copied/pasted from tutorial.md!  :-P

alimi commented on _installation/mac_os.md:
```diff
@@ -0,0 +1,118 @@
+
+1. install command-line tools (XCode)
+2. install ruby
+3. install rails
```
I prefer numbering all ordered list items with `1` because there's less stuff to worry about if the list is edited later.

```
1. install command-line tools (XCode)
1. install ruby
1. install rails
```

alimi reviewed (commented):
Nice job @solipet 👏🏾

I focused on markdown stuff when you looking over the changes. Things will probably change as the UI gets flushed out, but this is a great start.

alimi commented on _installation/mac_os.md:
```diff
@@ -0,0 +1,118 @@
+
+When you are installing ruby and ruby gems (libraries), your system will often need to compile code for you.  The compilers for this can be installed with XCode, the development tool suite for native mac os development.  If you think you might want to do some mac os x or iOS development, go ahead and install XCode (it's a LARGE download...), but if you just want to ignore this part and get on wit ...
+
+If you aren't sure if you need this, try typing `gcc` in a terminal window. (`gcc` is the name of the compiler and stands for GNU Compiler Collection):
```
On my new MacBook Pro (macOS Sierra 10.12.1), _any_ command that requires command line tools pops open a dialog asking to install them.

<img width="936" alt="screen shot 2016-12-20 at 9 41 10 pm" src="https://cloud.githubusercontent.com/assets/772925/21375679/15b3a468-c6fd-11e6-897e-dd5a43c61baf.png">

The dialog took me a couple of minutes to parse. "Install" will only install the command line tools, but "Get XCode" will install all the things. (We only need "Install").

solipet committed:
addressing Ali's comments on PR #4

alimi commented on _installation/mac_os.md:
```diff
@@ -0,0 +1,117 @@
+If you get the message, `-bash: gcc: command not found`, install the command-line tools with the command:
+
+{% highlight bash %}
+  > xcode-select --install
```
If you install command line tools via the dialog, you don't need to run `xcode-select --install`.

<img width="684" alt="screen shot 2016-12-20 at 9 55 12 pm" src="https://cloud.githubusercontent.com/assets/772925/21375918/02566566-c6ff-11e6-997a-b82a84674e37.png">

alimi commented on _installation/mac_os.md:
```diff
@@ -0,0 +1,117 @@
+To install rvm, open a terminal window and run the following commands (copied directly off of the rvm website):
+
+{% highlight bash %}
+  > gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
```
This command failed for me because my computer didn't already have gpg installed. To install it, I first installed homebrew. Then, I did `brew install gpg`.

alimi commented on _installation/mac_os.md:
```diff
@@ -0,0 +1,117 @@
+
+{% highlight bash %}
+  > gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
+\curl -sSL https://get.rvm.io | bash -s stable
```
At first glance, it wasn't clear that this was another thing I needed to run. Maybe all new lines should start with a ">"? I'm not sure - it might've just been difficult to read in the raw markdown file.

alimi replied:
Unrelated - I don't remember using gpg before, but this installs a lot of confidence...

```
gpg: Good signature from "Michal Papis (RVM signing) <mpapis@gmail.com>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
```
😏

solipet replied:
Ugh, that's unfortunate!  Those lines are copied directly off of rvm.io...

alimi commented on _installation/mac_os.md:
```diff
@@ -0,0 +1,117 @@
+\curl -sSL https://get.rvm.io | bash -s stable
+{% endhighlight %}
+
+Now that you have rvm installed, you can use it to install ruby.  By default, it installs the latest version, but if you want/need an older version, that's easy too.  For now, let's install the latest version (as of 12/3/2016), v2.3.1:
```
I'm a little confused by "By default, it installs the latest version"

Does that mean RVM ships with the latest version of Ruby? Or is does it mean`rvm install` with no options will install the latest version of ruby? (I _think_ it's the latter)

alimi commented on _installation/mac_os.md:
```diff
@@ -0,0 +1,117 @@
+Now that ruby is installed, installing rails is simple.  Just run the following command:
+
+{% highlight bash %}
+  > gem install rails 5.0.0
```
I've noticed the past couple of years that when we do `gem install rails`, we end up installing a ton of documentation. The attendees don't need the docs installed locally, and installing them takes a while.

I wonder if we can get away with doing `gem install rails 5.0.0 --no-document` instead.

http://blog.honeybadger.io/how-to-globally-disable-rdoc-and-ri-during-gem-installs/

alimi reviewed (commented):
I went through the install guide on a clean MacBook Pro (macOS Sierra 10.12.1) and left some feedback on my experience.

Overall, it went pretty smoothly but I may be a bad judge - this isn't my first rodeo.

![102-mommie-dearest-quotes](https://cloud.githubusercontent.com/assets/772925/21376836/3b630f92-c706-11e6-81a6-bb9da97c8c05.gif)

solipet committed:
Updates based upon Ali's installation test.

solipet committed:
reverted back to ruby 2.3.1 (not sure why I bumped it up)

solipet commented:
Ali, I made edits based on your installation experience.  Let me know if you want anything else changed (or if I missed anything)

solipet committed:
changed terminal prompt from '>' to '$'

solipet committed:
Merge branch 'master' into doug-installation-guide

solipet committed:
Got things working with the css from master.
Installation guide now includes a custom header that includes a css file
for overrides.

solipet commented:
OK, now the mac installation guide is integrated with the latest structure on master.  I added an installation.css file to override some css on the highlight blocks to make them look better for the guide as it's currently written.  Also set images to 70% (arbitrary) width and centered.
